services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -339,10 +342,10 @@ public Stream<UserRepresentation> getUsers(
                     attributes.putAll(searchAttributes);
 
                     return searchForUser(attributes, realm, userPermissionEvaluator, briefRepresentation, firstResult,
-                            maxResults, true);
+                            maxResults, includeServiceAccounts);
```

Comment:
@BbIKTOP We should keep backward compatibility. 

This is the only place we want to include service accounts and it is clearly targeted for querying regular users. In addition to that, not leveraging the index when querying users or service accounts is becoming a problem for those with a huge number of users (regular and SA) in the database.

They are really different things.

I understand it seems like a simple fix but the core issue is popping up quite often when users and service accounts are managed like the same.

I would rather add a specific endpoint to ClientsResource to query/search service accounts if we want to break backward compatibility.

Otherwise, we need to default to `true` here when querying by username and other parameters.

Reply:
Well, actually this solution was a "quick fix" which ruined consistency. Now at least two endpoints ain't consistent - /users with and without filtering as well as users/count. So, consistent solution is to align at least these. Also, I don't see any properties in the returned objects that allows to tell is this account service or not.
Sorry, I see it for the first time, there are definitely more details i don't know about yet. Although it's obvious that current workaround with this "true" brought inconsistency. I suppose, defaulting it to "true" will solve the problem. Should I change it?
Also, it would be a good idea to add some attribute to distinguish returned accounts (whether account is service or not)

Reply:
No problem and thanks for your first contribution.

The `includeServiceAccounts` should only default to `true` in the line above. The other places should be passing `false` as it stands today.

We also need to:

* Update the [Java Admin Client](https://github.com/keycloak/keycloak/blob/main/integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/UsersResource.java) to include the new parameter
* Create a test here https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UsersTest.java

Does it work for you?

About what I mentioned before, I'll create an issue to get feedback about how we can improve how we are dealing with SAs from the User API.

Reply:
Well, so, should it be set to the parameter's value if one exists in the request, and defaulted to the current values if there's none?
Unfortunately I'm currently completely out of free time, need to repair my house shelled by russians,  and also working hard on my work projects. I understood the task and will return back to it as soon as I could. Thank you!

Reply:
Sad to hear that. I wish you all the best.

I'll take it over if you don't mind.